subsys/bluetooth/controller/ll_sw/ull_adv_aux.c

```diff
@@ -595,10 +595,49 @@ uint8_t ll_adv_aux_set_remove(uint8_t handle)
 
 uint8_t ll_adv_aux_set_clear(void)
 {
-	/* TODO: reset/release all adv set primary channel and  Aux channel
-	 * PDUs
-	 */
-	return 0;
+	struct ll_adv_set *adv;
+	struct lll_adv *lll;
+	uint8_t handle;
+
+	for (handle = 0; handle < BT_CTLR_ADV_SET; ++handle) {
+		/* Get the advertising set instance */
+		adv = ull_adv_is_created_get(handle);
+
+		if (!adv) {
+			continue;
+		}
+
+		if (adv->is_enabled) {
```

Comment:
I know the spec is unclear about the behaviour here, but what happens now is that we release advertiser sets up to the first one that is enabled, and do not release the remaining ones.
So for example if we have 3 instances (1, 2 and 3) and nr. 2 is still enabled we remove nr. 1, and return error-code BT_HCI_ERR_CMD_DISALLOWED, but instance nr. 3 remains untouched.
Wouldn't it be more consistent to either remove all instances (that are not enabled), or to keep all instances?

Reply:
We should remove all, or none. Which implies, 2 step process, of check for all sets being disabled, and then remove all. ` ll_adv_aux_set_remove` be refactored into 2 smaller functions and reused here in 2 loops (check loop, and remove loop).

Reply:
Changed to disable all enabled sets, return error if one or more was enabled.